pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -234,8 +236,17 @@ func internalValidateServiceInstanceUpdateAllowed(new *sc.ServiceInstance, old *
 // ValidateServiceInstanceUpdate validates a change to the Instance's spec.
 func ValidateServiceInstanceUpdate(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
 	allErrs := field.ErrorList{}
+
 	allErrs = append(allErrs, internalValidateServiceInstanceUpdateAllowed(new, old)...)
 	allErrs = append(allErrs, internalValidateServiceInstance(new, false)...)
+
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(new.Spec.ExternalServiceClassName, old.Spec.ExternalServiceClassName, field.NewPath("spec").Child("externalServiceClassName"))...)
```

Comment:
We also need to check that plan hasn't changed if plan updates are not supported (`ServiceClass.Spec.PlanUpdatable`==false).
If it's costly to validate it there, we can check it before sending an update request to the broker.

Reply:
I am not sure about this, but it is my understanding that the validation should not be looking at other resources. The validation should only consider the resource being validated. There is an admission controller [1] that blocks updates to `ExternalServicePlanName` when the `PlanUpdatable` field of the service class is false.

[1] https://github.com/kubernetes-incubator/service-catalog/blob/master/plugin/pkg/admission/serviceplan/changevalidator/admission.go

Reply:
@staebler didn't know we have an admission controller for that. then there is no need for checking it again, of course.